src/compiler/factory.ts

```diff
@@ -214,6 +223,68 @@ namespace ts {
         return <BooleanLiteral & Token<SyntaxKind.FalseKeyword>>createSynthesizedNode(SyntaxKind.FalseKeyword);
     }
 
+    // Modifiers
+
+    export function createAbstractModifier() {

```

Comment:
Public API. It's very odd to have one but not the others.

Reply:
Maybe we should add these to services since they don't need to be in tsc?

Reply:
I don't think we've usually inlined a single-use sensible function (and we have a lot of internal single-use functions) and copied it into services for a tiny possible benefit to tsc elsewhere. It doesn't really make sense to split up the factory functions like that, from an organizational standpoint. Plus as you said, they'll almost certainly get inlined by the JIT if it matters, and function bodies are parsed lazily in the engine anyway, so any tiny parse overhead is deferred anyway. Tbqh, if you were into that I'd say we should have a tool for automatically inlining things based on a known heuristic (like the maximum bytecode size before the containing function can be inlined, for example), rather than speculating on it, and making the API harder to maintain in the process.

Reply:
Heck, we should probably just be running a dead code elimination tool on tsc.

Reply:
I just don't feel that these are worth it. If we are purely adding them as a convenience then we should have one for every token you can create that we emit, and we don't do that for things like `QuestionToken`, `EqualsGreaterThanToken`, etc.

Reply:
Those have `token` in the name, which leads one to believe `createToken` is the correct way to go (and should appear in quickinfo if you're searching blindly, which, given our lack of docs, is what will probably be done), while these show up in quick info when you search with `modifier`, which is the name of the type where you'd use them. These also have an implied correct location in the AST (modifier positions), while a "question token" is a bit ambiguous - there's no obvious question token locations (there're all called out as `.questionToken` on a parent node) and there's little in common between where they appear.

Pretty much just that while `Modifier`s are effectively just tokens in the implementation, they carry some extra semantic baggage that makes it more obvious how to get a hold of one of them if they actually have individual public functions.

Reply:
I've merged them all into one method